### PR TITLE
Matching alternatives count output

### PR DESCRIPTION
**Pull request by gardster**
# Issue

https://github.com/Project-OSRM/osrm-backend/issues/3508

Possible way to determine degree of "finality" of the solution is how many ways that points to final HMM state lefts in the current layer. If there is only one candidate, that has link with final candidates set, than we can't change this candidate in future, even we obtain additional points after the last one. So we can save all matching results for points without alternatives and match only a part of the track when we obtain additional points. 

## Tasklist
 - [x] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: include/engine/routing_algorithms/map_matching.hpp [hunk 1]
```diff
@@ -370,12 +371,8 @@ class MapMatching final : public BasicRoutingInterface<DataFacadeT, MapMatching<
             std::deque<std::pair<std::size_t, std::size_t>> reconstructed_indices;
             while (parent_timestamp_index > sub_matching_begin)
             {
-                if (model.breakage[parent_timestamp_index])
```
Comment (TheMarex) [inline include/engine/routing_algorithms/map_matching.hpp]: This seems unlreated, why was this removed?

Comment (gardster) [inline include/engine/routing_algorithms/map_matching.hpp]: It can lead to infinite loop. We don't change `parent_timestamp_index` or `sub_matching_begin` here, but continue to the while statement.

**Comment by TheMarex** [comment]
I don't have time for an in-depth review right now, but I would like to see this in 5.7.0.

One thing I think is confusing is that the number of alternatives includes the currently selected route. "no alternatives" -> alternatives=0 would be the more intuitive interpretation. You probably need to port this to the `src/engine/routing_algorithms/map_matching.cpp` file since we shuffled code around.

**Comment by gardster** [comment]
Ok. I port a code to the current master and decrease alternatives_count value.

**Comment by TheMarex** [comment]
Finally had time to look at this again: Looks great. 👍 Merging.
